cl/search/signals.py

```diff
@@ -0,0 +1,81 @@
+from cl.lib.custom_es_signal_processor import ESSignalProcessor
+from cl.search.documents import ParentheticalGroupDocument
+from cl.search.models import (
+    Citation,
+    Docket,
+    Opinion,
+    OpinionCluster,
+    OpinionsCited,
+    Parenthetical,
+    ParentheticalGroup,
+)
+
+pa_fields_mapping = {
+    "save": {
+        Docket: {
+            "opinion__cluster__docket": {
+                "docket_number": ("docketNumber", "raw"),
+                "court_id": ("court_id", "raw"),
+            }
+        },
+        Opinion: {
+            "opinion": {
+                "author_id": ("author_id", "raw"),
+                "cluster_id": ("cluster_id", "raw"),
+                "extracted_by_ocr": ("opinion_extracted_by_ocr", "raw"),
+            },
+        },
+        OpinionCluster: {
+            "representative__describing_opinion__cluster": {
+                "slug": ("describing_opinion_cluster_slug", "raw")
+            },
+            "opinion__cluster": {
+                "case_name": ("caseName", "raw"),
+                "citation_count": ("citeCount", "raw"),
+                "date_filed": ("dateFiled", "raw"),
+                "slug": ("opinion_cluster_slug", "raw"),
+                "docket_id": ("docket_id", "raw"),
+                "judges": ("judge", "raw"),
+                "nature_of_suit": ("suitNature", "raw"),
+                "precedential_status": (
+                    "status",
+                    "display",
+                ),  # This should be returned as display()
```

Comment:
Could you get rid of all the `raw` values by making this `get_precedential_status_display`, and having code that tries to invoke the attribute, and failing that, tries to `get` it?

If you did that, this whole thing would get simpler. For example, you could make each thing a `dict[str, str]` instead of `dict[str, tuple[str, str]]`:

```
"opinion__cluster": {
    "case_name": "caseName",
    "citation_count":  "citeCount",
    "date_filed": "dateFiled",
    "slug": "opinion_cluster_slug",
    "docket_id": "docket_id",
    "judges": "judge",
    "nature_of_suit": "suitNature",
    "get_precedential_status_display": "status",
}
```

I also worry a bit about how this reproduces so much of the parenthetical elastic search document object, but perhaps that's unavoidable?